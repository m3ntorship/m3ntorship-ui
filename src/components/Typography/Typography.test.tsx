import React from 'react';
import * as renderer from 'react-test-renderer';

import { Typography } from '.';

describe('Typography', () => {
  it('Should render h1', () => {
    const tree = renderer.create(
      <Typography variant="h1">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h2', () => {
    const tree = renderer.create(
      <Typography variant="h2">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h3', () => {
    const tree = renderer.create(
      <Typography variant="h3">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h4', () => {
    const tree = renderer.create(
      <Typography variant="h4">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h5', () => {
    const tree = renderer.create(
      <Typography variant="h5">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h6', () => {
    const tree = renderer.create(
      <Typography variant="h6">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render hero', () => {
    const tree = renderer.create(
      <Typography variant="hero">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render body1', () => {
    const tree = renderer.create(
      <Typography variant="body1">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render body2', () => {
    const tree = renderer.create(
      <Typography variant="body2">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render subtitle', () => {
    const tree = renderer.create(
      <Typography variant="subtitle">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render caption', () => {
    const tree = renderer.create(
      <Typography variant="caption">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render disclaimer', () => {
    const tree = renderer.create(
      <Typography variant="disclaimer">Lorem ipsum dolor sit amet.</Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render div with h1 styles', () => {
    const tree = renderer.create(
      <Typography variant="h1" component="div">
        Lorem ipsum dolor sit amet.
      </Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h1 with right alignment', () => {
    const tree = renderer.create(
      <Typography variant="h1" align="right">
        Lorem ipsum dolor sit amet.
      </Typography>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Should render h1 with className bg-red-200', () => {
    const tree = renderer.create(
      <Typography variant="h1" className="bg-red-200">
        Lorem ipsum dolor sit amet.
      </Typography>
    );
    expect(tree).toMatchSnapshot();
  });
});
