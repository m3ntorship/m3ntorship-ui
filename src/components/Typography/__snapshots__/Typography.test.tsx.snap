// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Should render body1 1`] = `
<p
  className="font-inter text-dgrey-800 text-left body1"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render body2 1`] = `
<p
  className="font-inter text-dgrey-800 text-left body2"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render caption 1`] = `
<p
  className="font-inter text-dgrey-800 text-left caption"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render disclaimer 1`] = `
<p
  className="font-inter text-dgrey-800 text-left disclaimer"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render div with h1 styles 1`] = `
<div
  className="font-inter text-dgrey-800 text-left h1"
>
  Lorem ipsum dolor sit amet.
</div>
`;

exports[`Typography Should render h1 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h1"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h1 with className bg-red-200 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h1 bg-red-200"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h1 with right alignment 1`] = `
<p
  className="font-inter text-dgrey-800 text-right h1"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h2 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h2"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h3 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h3"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h4 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h4"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h5 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h5"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render h6 1`] = `
<p
  className="font-inter text-dgrey-800 text-left h6"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render hero 1`] = `
<p
  className="font-inter text-dgrey-800 text-left hero"
>
  Lorem ipsum dolor sit amet.
</p>
`;

exports[`Typography Should render subtitle 1`] = `
<p
  className="font-inter text-dgrey-800 text-left subtitle"
>
  Lorem ipsum dolor sit amet.
</p>
`;
